File: LeetCode/SortByParity/3260.ts
function sortArrayByParity(nums: number[]): number[] {
  let left = 0;
  let right = nums.length - 1;

  while (left <= right) {
    if (nums[left] % 2 != 0) {
      while (nums[right] % 2 != 0 && right > left) {
        right--;
      }
      //swap
      let temp = nums[left];
      nums[left] = nums[right];
      nums[right] = temp;
    }
    left++;
  }
  return nums;
}

File: LeetCode/SortByParity/3260.cs
using System;
namespace ArrayParity{
    class Program{
        public int[] SortArrayByParity(int[] nums) {
        int left = 0;
        int right = nums.Length - 1;

        while(left<=right){
            if(nums[left]%2 != 0){
                while(nums[right]%2 != 0 && left < right){
                    right--;
                }
                //swap
                int temp = nums[left];
                nums[left] = nums[right];
                nums[right] = temp;
            }
            left++;
        }
        return nums;
        }
        static void Main(){

        }
    }
}